contracts/stellar-interchain-token-service/src/token_id.rs

```diff
@@ -2,11 +2,35 @@ use stellar_axelar_std::address::AddressExt;
 use stellar_axelar_std::xdr::ToXdr;
 use stellar_axelar_std::{Address, BytesN, Env};
 
+use crate::error::ContractError;
+use crate::storage;
+
 const PREFIX_CANONICAL_TOKEN_SALT: &str = "canonical-token-salt";
 const PREFIX_INTERCHAIN_TOKEN_SALT: &str = "interchain-token-salt";
+const PREFIX_CUSTOM_TOKEN_SALT: &str = "custom-token-salt";
 /// This prefix is used along with a salt to generate the token ID
 const PREFIX_TOKEN_ID: &str = "its-interchain-token-id";
 
+/// A newtype wrapper that enforces token registration checks have been performed.
+/// This prevents calling `deploy_token_manager` without first checking that the token ID
+/// is not already registered.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct UnregisteredTokenId(BytesN<32>);
+
+impl UnregisteredTokenId {
+    /// Create a new UnregisteredTokenId. This should only be called after verifying
+    /// that the token ID is not already registered.
+    pub(crate) const fn new(token_id: BytesN<32>) -> Self {
```

Comment:
```suggestion
    const fn new(token_id: BytesN<32>) -> Self {
```
This should be made private now that the ensure method has been moved here

Reply:
make sense, updated!